src/contexts/index.ts

```diff
@@ -39,12 +39,24 @@ export type SetState = (
   callback?: () => void,
 ) => void;
 
-export type DocStateContextProps = {
+export type BackBehaviorElements = {
+  add: (elements: Element[]) => void;
+  get: () => Element[];
+  remove: (elements: Element[]) => void;
+};
```

Comment:
Instead of defining this as a `type`, I think it might be worth defining it as a class that will encapsulate the actual logic for these methods:

```suggestion
export class BackBehaviorElements {
  add: (elements: Element[]) => void = {
    ...
  },
  ...
```

then it can simply be instantiated when needed.

Reply:
Done.